### ISSUE TITLE
AWS Europe London (eu-west-2) support

### ISSUE THREAD

**Issue by DavidGarciaCat** [liip/LiipImagineBundle#861]
Hello,

Just want to confirm if the bundle has already support to upload and generate thumbnails using a London's bucket.

Apparently, when I check an uploaded file through S3 console I get different base URLs:
- Ireland: https://s3-eu-west-1.amazonaws.com/ireland-bucket-name/
- London: https://s3.eu-west-2.amazonaws.com/london-bucket-name/

I have some S3 bucket parameters on my Symfony configuration file, but looks like is not enough to change these parameters, and I cannot load a file that is hosted in London.

Am I doing something wrong? Is this bundle offering support for London buckets?

Thanks,

**Comment by robfrawley** [comment]
@DavidGarciaCat Can you please post your complete config (with credentials censored, of course) and services definitions?

**Comment by DavidGarciaCat** [comment]
Hey @robfrawley 

Thanks for the reply - please find below my configuration details:

`app/config/parameters.yml`
```yml
parameters:
    # Other parameters here
    aws.key: MyAwesomeAccessKey
    aws.secret: MyAwesomeSecretKey
    aws.s3.bucket_name: MyAwesomeBucketName
```

`app/config/imagine.yml`
```yml
liip_imagine:
    loaders: ~
    data_loader: stream.photo_storage
    cache: amazon_s3
    filter_sets:
        # My awesome filter settings are here
        # There are 16 - Do you need them too?
```

`app/config/config.yml`
```yml
imports:
    # Other imported configuration files
    - { resource: imagine.yml }

# Main configuration here
```

Also, I have implemented a custom Amazon S3 Resolver as described [here](https://symfony.com/doc/current/bundles/LiipImagineBundle/cache-resolver/aws_s3.html#configuration) but it is an XML file instead of YAML (please note it was not built by me, another developer did it). Also, apparently, there is no reference to `liip_imagine.resolvers` key on config files as you have seen, but in the end I think that should not be (for now) a blocker due we provide the Access/Secret Key and the Bucket name, so the URL is auto-generated (right?):

`src/MyAwesomeNameSpace/StorageBundle/Resources/config/services.xml`
```xml
<?xml version="1.0" ?>

<container xmlns="http://symfony.com/schema/dic/services"
           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">

    <parameters>
        <parameter key="acme_storage.amazon_s3.class">AmazonS3</parameter>
        <parameter key="acme_storage.resolver.class">Liip\ImagineBundle\Imagine\Cache\Resolver\AmazonS3Resolver</parameter>
    </parameters>

    <services>
        <service id="acme_storage.amazon_s3" class="%acme_storage.amazon_s3.class%">
            <argument type="collection">
                <argument key="key">%aws.key%</argument>
                <argument key="secret">%aws.secret%</argument>
            </argument>
        </service>

        <service id="acme_storage.liip_imagine.binary.loader.stream.photo_storage"
                 class="%liip_imagine.binary.loader.stream.class%">
            <argument>gaufrette://photo_storage/</argument>
            <tag name="liip_imagine.binary.loader" loader="stream.photo_storage" />
        </service>

        <service id="acme_storage.imagine.cache.resolver.amazon_s3"
                 class="%acme_storage.resolver.class%">
            <argument type="service" id="acme_storage.amazon_s3" />
            <argument>%aws.s3.bucket_name%</argument>
            <argument>public-read</argument>
            <argument type="collection">
                <argument key="https">true</argument>
                <argument key="torrent">false</argument>
            </argument>
            <tag name="liip_imagine.cache.resolver" resolver="amazon_s3" />
        </service>

        <!--
        This configuration file contains another 2 services, that are custom uploaders.
        They are uploading files as expected, and I have access to them via AWS S3 console.
        -->
    </services>

</container>
```

After a quick check on my local source code, I think this is all that I have. Hopefully it can help to find a solution for my problem retrieving the images and generating the expected thumbnails.

Regards,

**Comment by antoligy** [comment]
Nothing stands out as being wrong there, would you be able to check to see if any cache files are being generated in the London bucket?
Also just to check, are you using multiple regions/buckets in parallel?  (Misbehaving code or a misconfiguration here might explain the issue)

**Comment by DavidGarciaCat** [comment]
Hello @antoligy 

For now, I cannot see any cached or generated thumbnail file in London's bucket. Also, I have checked my local cache to be sure it won't be affecting (deleting browser cache & cookies), I double-checked cache software (memcached) was not running (it wasn't), etcetera.

Yes, currently the company that I work for has several buckets in Ireland. We've planned to move the server and storage infrastructure to London (business reasons), so we've started to make these changes, and the 1st bucket we planned to move was causing this URL error.

Also, I have been testing some things on the new London's EC2 instances, and `s3cmd` was working fine with London's bucket from London's server, however Ireland instance was not able to connect to the new London's bucket. My IAM User has no restrictions working with S3 (no specific bucket name, no specific regions), so policies shouldn't be a problem here.

Any other idea? I'm still concerned about the URLs when I check the uploaded files via AWS console, due looks like each bucket region provides a different url (as described above):
- Ireland (`s3-eu-west-1`): https://s3-eu-west-1.amazonaws.com/ireland-bucket-name/...
- London (`s3.eu-west-2`): https://s3.eu-west-2.amazonaws.com/london-bucket-name/...

Thanks for your time guys,
